sensor-things-api-core/src/main/java/org/n52/sta/data/service/ObservationService.java

```diff
@@ -324,13 +325,53 @@ private void check(StaDataEntity observation) throws ODataApplicationException {
         }
     }
 
+    /**
+     * Handles updating the phenomenonTime field of the associated Datastream when Observation phenomenonTime is
+     * updated or deleted
+     */
+    private void updateDatastreamPhenomenonTimeOnObservationUpdate(
+            DatastreamEntity datastreamEntity, DataEntity<?> observation) {
+            if (observation.getPhenomenonTimeStart().compareTo(datastreamEntity.getPhenomenonTimeStart()) != 1
+                    || observation.getPhenomenonTimeEnd().compareTo(datastreamEntity.getPhenomenonTimeEnd()) != -1
+            ) {
+                // Setting new phenomenonTimeStart
+                Date newPhenomenonStart = getRepository()
+                        .findFirstByOrderBySamplingTimeStartAsc()
+                        .getPhenomenonTimeStart();
+                if (newPhenomenonStart != null) {
+                    datastreamEntity.setPhenomenonTimeStart(newPhenomenonStart);
+                } else {
+                    datastreamEntity.setPhenomenonTimeStart(null);
+                    datastreamEntity.setPhenomenonTimeEnd(null);
+                }
+                // Setting new phenomenonTimeEnd
+                Date newPhenomenonEnd = getRepository()
+                        .findFirstByOrderBySamplingTimeEndDesc()
+                        .getPhenomenonTimeStart();
+                if (newPhenomenonEnd != null) {
+                    datastreamEntity.setPhenomenonTimeEnd(newPhenomenonEnd);
+                } else {
+                    datastreamEntity.setPhenomenonTimeStart(null);
+                    datastreamEntity.setPhenomenonTimeEnd(null);
+                }
+                datastreamRepository.save(datastreamEntity);
+            }
+    }
+
```

Comment:
What does _findFirstByOrderBySamplingTimeStartAsc_ and _findFirstByOrderBySamplingTimeEndDesc_ do?
I think they querying the first/last observation of all observations and not for the Datastream. 
On the other hand, why not updating the phenomenonTime with the time values from the observation (DataEntity)?

Reply:
>On the other hand, why not updating the phenomenonTime with the time values from the observation (DataEntity)?  

The Datastream's phenomenonTime can only be updated with the time values from the observation if it's sure that these value would update the phenomenonTime. There are two cases this would happen:
1) The updated startTime of the observation is before the phenomenonStartTime of the datastream
2) The updated endTime of the observation is after the phenomenonEndTime of the datastream

In all other cases the phenomenonStartTime and phenomenonEndTime have to be determined from the observations associated with the affected Datastream.

>What does findFirstByOrderBySamplingTimeStartAsc and findFirstByOrderBySamplingTimeEndDesc do?
I think they querying the first/last observation of all observations and not for the Datastream.

That should be fixed, so that the new phenomenonStartTime and phenomenonEndTime will be determined from the observations associated with the Datastream. Be sure, that the updated observation values or startTime and endTime are taken into account when doing so.